client/state/timezones/reducer.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TIMEZONES_RECEIVE,
+	TIMEZONES_REQUEST,
+	TIMEZONES_REQUEST_FAILURE,
+	TIMEZONES_REQUEST_SUCCESS
+} from 'state/action-types';
+import { timezonesSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+/**
+ * Collect and store timezones data
+ *
+ * @param  {Object} state - Current state
+ * @param  {Object} action - action payload
+ * @return {Object} updated state
+ */
+export const items = createReducer( {}, {
+	[ TIMEZONES_RECEIVE ]: ( state, {
+		rawOffsets,
+		timezonesByContinent,
+	} ) => ( { rawOffsets, timezonesByContinent } ),
+}, timezonesSchema );
+
+/**
+ * Track the timezones requesting process
```

Comment:
I suggest go ahead with the current approach. It means to keep the requesting state in the state-tree and work in a better approach to another PR and I think it's out of the scope of this one.
Also, keep in mind that there are following PRs waiting for this to continue ahead.